Webhook rework

AdiReich opened the pull request:
I implemented KrizBOT into AraBOT.
It is slightly improved version where we provide message instead of context and read bot avatar from self.

There is one thing that need to be added but I'm not sure how to do it in elegant way.
You will get error when bot attempts to create new webhook but there is no more space for it in channel

Discord limit per channel is 15 webhooks so it is rare but can happen

AdiReich committed:
modified:   arabot/core/bot.py
	modified:   arabot/modules/general.py

AdiReich committed:
message

tooruu commented on arabot/modules/general.py:
```diff
@@ -8,15 +8,8 @@
-    bold,
-)
+from arabot.utils import (CUSTOM_EMOJI_RE, AnyEmoji, AnyEmojis, AnyMember,
+                          AnyMemberOrUser, Twemoji, bold)
```
Not in scope of the PR

AdiReich committed:
Func.name change
Changed func. name
get_webhook >> fetch_webhook
fixed imports in arabot/modules/general.py

tooruu commented on arabot/core/bot.py:
```diff
@@ -155,6 +155,18 @@ def load_extensions(self) -> None:
+    async def get_webhook(self, name: str, msg: disnake.Message) -> disnake.Webhook | None:
+        reason = (
+            "Used for some functionality. It is safe to delete. Bot will recreate it when required."
+        )
```
Too long. No one will read it anyway.

AdiReich committed:
Co-authored-by: tooru <tooru@tooru.uk>

tooruu reviewed (approved):
LGTM!
